coalib/output/ConfWriter.py

```diff
@@ -89,15 +92,35 @@ def __write_key_val(self, keys, val):
 
         # Add escape characters as appropriate
         keys = [escape(key, chain(['\\'],
-                                  self.__key_value_delimiters,
+                                  self.__key_value_define_delimiters,
                                   self.__comment_separators,
                                   self.__key_delimiters,
                                   self.__section_override_delimiters))
                 for key in keys]
         val = escape(val, chain(['\\'], self.__comment_separators))
 
-        self.__file.write((self.__key_delimiter + ' ').join(keys) + ' ' +
-                          self.__key_value_delimiter + ' ' + val + '\n')
+        append_keys = []
+        for key in keys:
+            if (section.defaults and
+                    (key in section.defaults and
+                     (val.startswith(str(section.defaults[key])+',')))):
+                append_keys.append(key)
+
+        if append_keys:
+            def_val_list = str(section.defaults[append_keys[0]]).split(
+                self.__key_delimiter)
+            append_val = (self.__key_delimiter + ' ').join(
+                [v.strip() for v in val.split(self.__key_delimiter)
+                 if v not in def_val_list])
+            self.__file.write((self.__key_delimiter + ' ').join(append_keys) +
+                              ' ' + self.__key_value_append_delimiter + ' ' +
+                              append_val + '\n')
+            keys = [key for key in keys if key not in append_keys]
```

Comment:
I dislike this a bit, can't we just compile a second list in an else in 107 maybe? Or work with sets?